source/DataHolder/AMDSGenericFlatArrayDataHolder.h

```diff
@@ -10,25 +10,31 @@ Q_OBJECT
 	AMDSLightWeightGenericFlatArrayDataHolder(AMDSDataTypeDefinitions::DataType dataType = AMDSDataTypeDefinitions::Double, quint32 size = 2, QObject *parent = 0);
 	virtual ~AMDSLightWeightGenericFlatArrayDataHolder();
 
-	virtual inline QList<AMDSAxisInfo> axes() const = 0;
-	virtual inline quint8 rank() const  = 0;
-	/// Returns the length of all the axes
-	virtual inline AMDSnDIndex size() const  = 0;
-	/// Returns the length of the specified axis \c axisId.  (\c axisId is assumed to be >= 0 and < rank().)
-	virtual inline quint32 size(int axisId) const  = 0;
-	/// Returns the number of points this measurement spans (A scalar value is "1" point, a 1D Detector is the same as its dimension, higher-D detectors are the products of their dimensions)
+	/// Implement the function to return the number of points this measurement spans (A scalar value is "1" point, a 1D Detector is the same as its dimension, higher-D detectors are the products of their dimensions)
 	virtual inline quint64 spanSize() const;
 
-	virtual inline bool data(AMDSFlatArray *outputValues) const;
+	/// Implement the PLUS operation of AMDSLightWeightDataHolder, which will plus the value of the two instances of AMDSLightWeightDataHolder and return the new instance
+	virtual AMDSDataHolder* operator +(AMDSDataHolder &dataHolder);
+	/// Implement the Division operation of AMDSLightWeightDataHolder: the value of the given handler will be divided by the given divisior
+	virtual AMDSDataHolder* operator /(quint32 divisor);
 
-	virtual inline void setData(AMDSFlatArray *inputValues);
+	/// implement the data function to read data from valueFlatArray_ and write data to the given outputArray
+	virtual inline bool data(AMDSFlatArray *outputArray) const { return valueFlatArray_.resetTargetArrayAndReplaceData(outputArray); }
+	/// implement the setData function to update valueFlatArray_ with given inputArray
+	virtual inline void setData(AMDSFlatArray *inputArray) { inputArray->copyDataToTargetArray(&valueFlatArray_); }
```

Comment:
@davidChevrier @dretrex yes, the original data() and setData() were confusing. I'm trying to make it clear here.
`resetTargetArrayAndReplaceData()` function calls `copyDataToTargetArray()` function. that's the reason we pass `outputArray` to `resetTargetArrayAndReplaceData()`. In this way, `outputArray` will carry the data.